File: RFI.API/Models/Poster.cs
namespace RFI.API.Models
{
    public class Poster
    {
        public int Id { get; set; }
        public int EventId { get; set; }  // Foreign key to Event
        public string Title { get; set; } = string.Empty;
        public string FileUrl { get; set; } = string.Empty;
        public string ThumbnailUrl { get; set; } = string.Empty;
        public long FileSize { get; set; }  // in bytes
        public int DownloadCount { get; set; } = 0;
        public DateTime UploadedAt { get; set; } = DateTime.UtcNow;

        // Navigation property
        public Event? Event { get; set; }
    }
}

File: rfi-frontend/src/types/Poster.ts
export interface Poster {
  id: number;
  title: string;
  description?: string;
  fileUrl: string;
  thumbnailUrl?: string;
  fileSize: number;
  downloadCount: number;
  uploadedAt: string;
  tags?: string;
}

export interface CreatePosterRequest {
  file: File;
  title: string;
  description?: string;
}
